lib/truffle/truffle/cext.rb

```diff
@@ -1422,7 +1422,7 @@ def rb_io_printf(out, args)
   end
 
   def rb_io_print(out, args)
-    out.print(*args)
+    Truffle::IOOperations.print out, args, nil
```

Comment:
How did this end up in an infinite recursion?

Reply:
`rb_io_print` > calls `print` in GzipWriter > defined with `rb_gzwriter_print` in C > aliased to `rb_io_print`

Reply:
Oh, wow, interesting. At least the tests should catch it if we miss that.

Reply:
It was caught by the MRI tests in the same commit, it was not exercised in spec though.

Reply:
That's OK.